src/main/java/hudson/plugins/spotinst/jobs/SpotinstSyncGroupsOwner.java

```diff
@@ -0,0 +1,176 @@
+                           .map(cloud -> (BaseSpotinstCloud) cloud).filter(BaseSpotinstCloud::isActive)
+                           .collect(Collectors.toList());
+        }
+        else {
+            retVal = new ArrayList<>();
+        }
+
+        return retVal;
+    }
+
+    private void HandleCloudsChangeFromLastRun(List<BaseSpotinstCloud> activeClouds) {
+        Set<GroupLockKey> previousActiveGroups = getGroupsFromLastRun();
+        Set<GroupLockKey> currentActiveGroups  = getGroups(activeClouds);
+
+        previousActiveGroups.removeAll(currentActiveGroups);
+
+        deallocateGroups(previousActiveGroups);
+        groupsFromLastRun = currentActiveGroups;
+    }
+
+    private static Set<GroupLockKey> getGroups(List<BaseSpotinstCloud> clouds) {
+        Set<GroupLockKey> retVal =
+                clouds.stream().map(baseCloud -> new GroupLockKey(baseCloud.getGroupId(), baseCloud.getAccountId()))
+                      .collect(Collectors.toSet());
+        return retVal;
+    }
+
+    private void deallocateGroups(Set<GroupLockKey> groupsNoLongerExist) {
+        for (GroupLockKey groupKeyNoLongerExists : groupsNoLongerExist) {
+            String  groupId       = groupKeyNoLongerExists.getGroupId(), accountId =
+                    groupKeyNoLongerExists.getAccountId();
+            boolean isActiveCloud = StringUtils.isNotEmpty(groupId) && StringUtils.isNotEmpty(accountId);
+
+            if (isActiveCloud) {
+                ILockRepo groupControllerLockRepo = RepoManager.getInstance().getLockRepo();
+                ApiResponse<String> lockGroupControllerResponse =
+                        groupControllerLockRepo.getGroupControllerLock(groupId, accountId);
+
+                if (lockGroupControllerResponse.isRequestSucceed()) {
+                    String  lockGroupControllerValue  = lockGroupControllerResponse.getValue();
+                    String  controllerIdentifier      = SpotinstContext.getInstance().getControllerIdentifier();
+                    boolean isGroupBelongToController = controllerIdentifier.equals(lockGroupControllerValue);
+
+                    if (isGroupBelongToController) {
+                        deallocateGroup(groupKeyNoLongerExists, groupControllerLockRepo);
+                    }
+                    else {
+                        LOGGER.warn("Controller {} could not unlock group {} - already locked by Controller {}",
+                                    controllerIdentifier, groupId, lockGroupControllerValue);
+                    }
+                }
+                else {
+                    LOGGER.error("group unlocking service failed to get lock for groupId {}, accountId {}. Errors: {}",
+                                 groupId, accountId, lockGroupControllerResponse.getErrors());
+                }
+            }
+        }
+    }
+
+    private void deallocateGroup(GroupLockKey groupNoLongerExists, ILockRepo groupControllerLockRepo) {
```

Comment:
rename groupNoLongerExists to groupToUnlock , not sure that it right to pass groupControllerLockRepo

Reply:
Done